apiserver/testing/fakecharmstore.go

```diff
@@ -45,32 +49,37 @@ func (s *CharmStoreSuite) SetUpTest(c *gc.C) {
 			checkers.DeclaredCaveat("username", s.DischargeUser),
 		}, nil
 	})
-	s.Srv = charmstoretesting.OpenServer(c, s.Session, charmstore.ServerParams{
+	db := s.Session.DB("juju-testing")
+	params := charmstore.ServerParams{
+		AuthUsername:     "test-user",
+		AuthPassword:     "test-password",
 		IdentityLocation: s.discharger.Location(),
 		PublicKeyLocator: s.discharger,
+	}
+	handler, err := charmstore.NewServer(db, nil, "", params, charmstore.V4)
+	c.Assert(err, jc.ErrorIsNil)
+	s.handler = handler
+	s.Srv = httptest.NewServer(handler)
+	s.Client = csclient.New(csclient.Params{
+		URL:      s.Srv.URL,
+		User:     params.AuthUsername,
+		Password: params.AuthPassword,
 	})
+
```

Comment:
I agree with you. I'll add a card for abstracting out common code in a separate function. Good point thanks!